pkg/controller/rolloutblock/rolloutblock_controller.go

```diff
@@ -0,0 +1,622 @@
+package rolloutblock
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+
+	shipper "github.com/bookingcom/shipper/pkg/apis/shipper/v1alpha1"
+	clientset "github.com/bookingcom/shipper/pkg/client/clientset/versioned"
+	shipperinformers "github.com/bookingcom/shipper/pkg/client/informers/externalversions"
+	shipperlisters "github.com/bookingcom/shipper/pkg/client/listers/shipper/v1alpha1"
+	shippererrors "github.com/bookingcom/shipper/pkg/errors"
+	stringUtil "github.com/bookingcom/shipper/pkg/util/string"
+)
+
+const (
+	AgentName = "rolloutblock-controller"
+
+	// maxRetries is the number of times an RolloutBlock will be retried before
+	// we drop it out of the rolloutblock workqueue. The number is chosen with
+	// the default rate limiter in mind. This results in the following backoff
+	// times: 5ms, 10ms, 20ms, 40ms, 80ms, 160ms, 320ms, 640ms, 1.3s, 2.6s,
+	// 5.1s, 10.2s.
+	maxRetries = 11
+)
+
+// Controller is a Kubernetes controller that updates rolloutblock objects.
+// The RolloutBlock objects' main objective is to block application and releases rollout during
+// an outage.
+//
+// RolloutBlock Controller has one primary workqueues: a rolloutblock updater
+// object queue.
+type Controller struct {
+	shipperClientset clientset.Interface
+	recorder         record.EventRecorder
+
+	applicationLister  shipperlisters.ApplicationLister
+	applicationsSynced cache.InformerSynced
+
+	releaseLister  shipperlisters.ReleaseLister
+	releasesSynced cache.InformerSynced
+
+	rolloutBlockLister shipperlisters.RolloutBlockLister
+	rolloutBlockSynced cache.InformerSynced
+
+	rolloutblockWorkqueue workqueue.RateLimitingInterface
+}
+
+// NewController returns a new RolloutBlock controller.
+func NewController(
+	shipperClientset clientset.Interface,
+	informerFactory shipperinformers.SharedInformerFactory,
+	recorder record.EventRecorder,
+) *Controller {
+	applicationInformer := informerFactory.Shipper().V1alpha1().Applications()
+	releaseInformer := informerFactory.Shipper().V1alpha1().Releases()
+	rolloutblockInformer := informerFactory.Shipper().V1alpha1().RolloutBlocks()
```

Comment:
would it make sense to keep camel case naming and change it to rolloutBlockInformer?